client/tests/integration/smartcontracts/validator_with_admin/src/lib.rs

```diff
@@ -1,21 +1,133 @@
-//! Runtime Validator which allows any operation done by `admin@admin` account.
-//! If authority is not `admin@admin` then `iroha_default_validator` is used.
-
+//! Runtime Validator which allows any [`TransferBox`] instructions by `admin@admin` account.
+//! If authority is not `admin@admin` then [`DefaultValidator`] is used as a backup.
 #![no_std]
-#![no_main]
+
+use iroha_validator::{
+    data_model::evaluate::{Error, ExpressionEvaluator},
+    parse,
+    prelude::*,
+};
 
 #[cfg(not(test))]
 extern crate panic_halt;
 
-use iroha_validator::{parse, prelude::*};
+struct CustomValidator(DefaultValidator);
+
+macro_rules! delegate {
+    ( $($visitor:ident($operation:ty)),+ $(,)? ) => { $(
+        fn $visitor(&mut self, authority: &AccountId, operation: $operation) {
+            self.0.$visitor(authority, operation);
+        } )+
+    }
+}
+
+// TODO:
+// 1. Make a macro that takes care of this boilerplate
+impl Visit for CustomValidator {
```

Comment:
we should make a `#[derive(DefaultValidator)]` or something similar that would replace non listed function with a default

Reply:
Actually sounds not so bad, I haven't thought about having macro to fallback to default implementation

Reply:
I'm going to close this, we'll see if this hinders usability and we'll find ways to make it simpler